core/src/main/scala/org/broadinstitute/dsde/rawls/entities/local/LocalEntityProvider.scala

```diff
@@ -28,79 +27,58 @@ import scala.util.{Failure, Success, Try}
+          } else if (!useCache) {
+            logger.info(s"entity statistics cache: miss (user request specified cache bypass) [${workspaceContext.workspaceIdAsUUID}]")
           }
-        }
-      }
-    }
-  }
-
-  private def opportunisticSaveEntityCache(metadata: Map[String, EntityTypeMetadata], dataAccess: DataAccess) = {
-    val entityTypesWithCounts: Map[String, Int] = metadata.map {
-      case (typeName, typeMetadata) => typeName -> typeMetadata.count
-    }
-    val entityTypesWithAttrNames: Map[String, Seq[AttributeName]] = metadata.map {
-      case (typeName, typeMetadata) => typeName -> typeMetadata.attributeNames.map(AttributeName.fromDelimitedName)
-    }
-    val timestamp: Timestamp = new Timestamp(workspaceContext.lastModified.getMillis)
-
-    dataAccess.entityCacheManagementQuery.saveEntityCache(workspaceContext.workspaceIdAsUUID,
-      entityTypesWithCounts, entityTypesWithAttrNames, timestamp).asTry.map {
-      case Success(_) => // noop
-      case Failure(ex) =>
-        logger.warn(s"failed to opportunistically update the entity statistics cache: ${ex.getMessage}. " +
-          s"The user's request was not impacted.")
-    }
+          // retrieve metadata, bypassing cache
+          calculateMetadataResponse(dataAccess, countsFromCache = false, attributesFromCache = false, rootSpan)
+        } else {
+          // system and request both have cache enabled. Check for existence and staleness of cache
+          cacheStaleness(dataAccess, rootSpan).flatMap {
+            case None =>
+              // cache does not exist - return uncached
+              logger.info(s"entity statistics cache: miss (cache does not exist) [${workspaceContext.workspaceIdAsUUID}]")
+              calculateMetadataResponse(dataAccess, countsFromCache = false, attributesFromCache = false, rootSpan)
+            case Some(0) =>
+              // cache is up to date - return cached
+              logger.info(s"entity statistics cache: hit [${workspaceContext.workspaceIdAsUUID}]")
+              calculateMetadataResponse(dataAccess, countsFromCache = true, attributesFromCache = true, rootSpan)
+            case Some(_) =>
+              // cache exists, but is out of date - check if this workspace has any always-cache feature flags set
+              cacheFeatureFlags(dataAccess, rootSpan).flatMap { flags =>
+                if (flags.alwaysCacheTypeCounts || flags.alwaysCacheAttributes) {
+                  rootSpan.putAttribute("alwaysCacheTypeCountsFeatureFlag", OpenCensusAttributeValue.booleanAttributeValue(flags.alwaysCacheTypeCounts))
+                  rootSpan.putAttribute("alwaysCacheAttributesFeatureFlag", OpenCensusAttributeValue.booleanAttributeValue(flags.alwaysCacheAttributes))
+                  logger.info(s"entity statistics cache: partial hit (alwaysCacheTypeCounts=${flags.alwaysCacheTypeCounts}, alwaysCacheAttributes=${flags.alwaysCacheAttributes}) [${workspaceContext.workspaceIdAsUUID}]")
```

Comment:
we are logging how stale the cache is inside of `EntityStatisticsCacheSupport .cacheStaleness()`. However, the place I really care about staleness is here, for the partial hits. When we intentionally serve stale data because one or more "alwaysCache*" feature flags are set, how stale is the data we are serving?